docs/search-v2.md

```diff
@@ -143,3 +143,206 @@ sleep(randint(1, 10))
+Suitable synonyms here include brand names or specific technical terms which won't tend to be localized.
+
+#### Updating
+
+In development synonyms can be updated very easily.
+Save your changes in the text file and run:
+
+```
+./manage.py es7_init --reload-search-analyzers
+```
+
+If no other changes were made to the index configurations,
+then this should apply successfully,
+and your already-indexed data will persist within the index and not require any indexing
+(because these synonyms are applied at query time).
+
+##### On production
+
+The synonym files need to be put in a bundle and uploaded to the Elastic Cloud.
+
+`cd` into the `elasticsearch` directory,
+and run the `create_bundle.sh` script to create a zip file with the appropriate directory structure.
+(You'll need to have `zip` installed for this command to work.)
+
+Then,
+either [create](https://www.elastic.co/guide/en/cloud/current/ec-custom-bundles.html#ec-add-your-plugin) an extension,
+or [update](https://www.elastic.co/guide/en/cloud/current/ec-custom-bundles.html#ec-update-bundles-and-plugins) the previously created extension.
+
+And in either case,
+[update the deployment configuration](https://www.elastic.co/guide/en/cloud/current/ec-custom-bundles.html#ec-update-bundles)
+with the custom extension.
+
+```eval_rst
+.. Note::
+  When updating the deployment after updating an already-existing extension,
+  Elastic Cloud may say that no changes are being applied.
+  That isn't true,
+  and through testing it seems like the extension is being updated,
+  and the search analyzers are being reloaded automatically.
+
+  From testing,
+  this seems to be the only approach to update and reload synonyms on the Elastic Cloud.
+  Updating the extension,
+  restarting the cluster and using the reload-search-analyzers command *won't* work.
+
+  Thankfully there's an open issue upstream to make managing synonyms easier with an API:
+  https://github.com/elastic/elasticsearch/issues/38523
+```
+
+### Character mappings
+
+Character mappings *cannot* be dynamically updated,
+this is because they're applied at index time.
+So any changes to a character mapping requires a re-index.
+
+Taking the addon example from above,
+we'd want to create character mappings like:
+
+```
+[
+  "add on => addon",
+  "add-on => addon",
+]
+```
+
+Post-tokenization `addon` doesn't contain an `on` token,
+so this is a suitable phrase to replace with.
+
+Unlike synonyms,
+character mappings are applied before any other part of the analysis chain,
+so space separated and hyphen-separated phrases need to both be added.
+
+In theory plural and conjugated forms of words also need to be specified,
+however in practice plural words tend to be covered by the singular replacement as well
+(e.g. "add on" is a substring in "add ons",
+so "add ons" is replaced by "addons")
+and there is marginal benefit to defining *every single* conjugation of a verb.
```

Comment:
Ideally we need some documentation explaining how to add synonyms that should be also directly linked from the project's readme file. Happy to delegate this to another PR but let's capture a ticket for it